File: Python/ContinuousSubArraySum.py
## Leetcode Problem - 523

def checkSubarraySum(nums,k):

    prefixSum = {0:-1}
    curSum = 0

    for i in range(len(nums)):
        curSum = (curSum + nums[i]) % k

        if curSum in prefixSum:
            if i - prefixSum[curSum]>1:
                return True
        else:
            prefixSum[curSum] = i

    return False



nums = [23,2,6,4,7]
k = 6

print(checkSubarraySum(nums,k))

File: Javascript/ContinuousSubArraySum.js
// Leetcode Problem - 523

var checkSubarraySum = function (nums, k) {
  const prefixSum = new Map();
  prefixSum.set(0, -1);

  let curSum = 0;

  for (let i = 0; i < nums.length; i++) {
    curSum = (curSum + nums[i]) % k;

    if (prefixSum.has(curSum)) {
      if (i - prefixSum.get(curSum) > 1) {
        return true;
      }
    } else {
      prefixSum.set(curSum, i);
    }
  }

  return false;
};
